addons/mrp/models/stock_rule.py

```diff
@@ -29,29 +30,34 @@ def _onchange_action_operation(self):
             domain = {'picking_type_id': [('code', '=', 'mrp_operation')]}
         return {'domain': domain}
 
-    @api.multi
-    def _run_manufacture(self, product_id, product_qty, product_uom, location_id, name, origin, values):
-        Production = self.env['mrp.production']
-        ProductionSudo = Production.sudo().with_context(force_company=values['company_id'].id)
-        bom = self._get_matching_bom(product_id, values)
-        if not bom:
-            msg = _('There is no Bill of Material found for the product %s. Please define a Bill of Material for this product.') % (product_id.display_name,)
-            raise UserError(msg)
-
-        # create the MO as SUPERUSER because the current user may not have the rights to do it (mto product launched by a sale for example)
-        production = ProductionSudo.create(self._prepare_mo_vals(product_id, product_qty, product_uom, location_id, name, origin, values, bom))
-        production.move_raw_ids = self.env['stock.move'].create(production._get_moves_raw_values())
-        production.action_confirm()
-        origin_production = values.get('move_dest_ids') and values['move_dest_ids'][0].raw_material_production_id or False
-        orderpoint = values.get('orderpoint_id')
-        if orderpoint:
-            production.message_post_with_view('mail.message_origin_link',
-                                              values={'self': production, 'origin': orderpoint},
-                                              subtype_id=self.env.ref('mail.mt_note').id)
-        if origin_production:
-            production.message_post_with_view('mail.message_origin_link',
-                                              values={'self': production, 'origin': origin_production},
-                                              subtype_id=self.env.ref('mail.mt_note').id)
+    @api.model
+    def _run_manufacture(self, procurements):
+        productions_values_by_company = defaultdict(list)
+        for procurement, rule in procurements:
+            bom = self._get_matching_bom(procurement.product_id, procurement.values)
+            if not bom:
+                msg = _('There is no Bill of Material found for the product %s. Please define a Bill of Material for this product.') % (procurement.product_id.display_name,)
+                raise UserError(msg)
+
+            productions_values_by_company[procurement.values['company_id'].id].append(rule._prepare_mo_vals(*procurement, bom))
```

Comment:
i would put company_id in the key of the namedtuple
using it in values make it appear opitionnal while it is required